x86/src/Data/Macaw/X86/ArchTypes.hs

```diff
@@ -738,6 +738,29 @@ data X86PrimFn f tp where
     :: !(f (BVType 128))
     -> X86PrimFn f (BVType 128)
 
+  SHA_sigma0
+    :: !(f (BVType 32))
+    -> X86PrimFn f (BVType 32)
+  SHA_sigma1
+    :: !(f (BVType 32))
+    -> X86PrimFn f (BVType 32)
+  SHA_Sigma0
+    :: !(f (BVType 32))
+    -> X86PrimFn f (BVType 32)
+  SHA_Sigma1
+    :: !(f (BVType 32))
+    -> X86PrimFn f (BVType 32)
+  SHA_Ch
+    :: !(f (BVType 32))
+    -> !(f (BVType 32))
+    -> !(f (BVType 32))
+    -> X86PrimFn f (BVType 32)
+  SHA_Maj
+    :: !(f (BVType 32))
+    -> !(f (BVType 32))
+    -> !(f (BVType 32))
+    -> X86PrimFn f (BVType 32)
```

Comment:
I'm not super thrilled with the short names, but I understand why they are that way (and that there aren't really better names). My only request is a pointer here to what these are.  Something like: These represent operations in the SHA256 instructions, see SHA256RNDS2, SHA256MSG1, and SHA256MSG2 in the ISA manual.

I also found concise definitions of `Ch`, `Maj`, and the sigma functions (https://web.archive.org/web/20130526224224/http://csrc.nist.gov/groups/STM/cavp/documents/shs/sha256-384-512.pdf on page 8) that might be nice to at least reference.